src/test/proxy/worker_proxy.py

```diff
@@ -1,253 +1,63 @@
-        response_1 = self._busy_worker.resume_job(self._uid_1)
-        self.assertFalse(response_1.is_success)
-        response_2 = self._busy_worker.resume_job(self._uid_2)
-        self.assertFalse(response_2.is_success)
-
-    @skipIfAbstract
-    def test_pause_resume_1(self) -> None:
-        response_1 = self._busy_worker.pause_job(self._uid_1)
-        self.assertTrue(response_1.is_success)
-        response_2 = self._busy_worker.resume_job(self._uid_2)
-        self.assertFalse(response_2.is_success)
-        response_3 = self._busy_worker.resume_job(self._uid_1)
-        self.assertTrue(response_3.is_success)
-
-    @skipIfAbstract
-    def test_pause_resume_2(self) -> None:
-        response_1 = self._busy_worker.pause_job(self._uid_2)
-        self.assertTrue(response_1.is_success)
-        response_2 = self._busy_worker.resume_job(self._uid_1)
-        self.assertFalse(response_2.is_success)
-        response_3 = self._busy_worker.resume_job(self._uid_2)
-        self.assertTrue(response_3.is_success)
-
-    @skipIfAbstract
-    def test_resume_unknown(self) -> None:
-        response_empty = self._empty_worker.resume_job(self._uid_unknown)
-        self.assertFalse(response_empty.is_success)
-        response_busy = self._empty_worker.resume_job(self._uid_unknown)
-        self.assertFalse(response_busy.is_success)
-
-
-class WorkerProxyDummyTest(AbstractWorkerProxyTest):
+from time import sleep
+from getpass import getuser
+from typing import List, Dict
+
+from ja.common.job import Job
+from ja.common.message.worker_commands.cancel_job import CancelJobCommand
+from ja.common.message.worker_commands.pause_job import PauseJobCommand
+from ja.common.message.worker_commands.resume_job import ResumeJobCommand
+from ja.common.message.worker_commands.start_job import StartJobCommand
+from ja.common.proxy.command_handler import CommandHandler
+from ja.common.proxy.ssh import SSHConfig, ISSHConnection
+from ja.server.proxy.proxy import WorkerProxyBase
+from test.proxy.abstract import AbstractWorkerProxyTest
+from test.proxy.worker_proxy_dummy import WorkerProxyDummy
+from test.ssh.test_ssh_dummy import SSHConnectionDummy
+
+
+class WorkerProxy(WorkerProxyBase):
+    """
+    Implementation of the proxy for the worker client that uses SSHConnectionDummy dor testing
+    """
+
+    def _get_ssh_connection(self, ssh_config: SSHConfig) -> ISSHConnection:
+        return SSHConnectionDummy(socket_path="./dummy_socket_%s" % ssh_config.hostname)
+
+
+class WorkerCommandHandlerDummy(CommandHandler):
+
+    def __init__(self, socket_path: str, jobs: List[Job]):
+        super().__init__(socket_path=socket_path)
+        self._worker_proxy_dummy = WorkerProxyDummy(uid=None, jobs=jobs)
+
+    def _process_command_string(
+            self, command_dict: Dict[str, object], type_name: str, username: str) -> Dict[str, object]:
+        assert username == getuser()
+        if type_name == "StartJobCommand":
```

Comment:
To avoid these, can we not dynamically import modules and classes based on their name?

Reply:
That sounds like it would be more work because you would need to get MyPy to accept it.

Reply:
Just cast?